src/Statement/StatementList.php

```diff
@@ -282,4 +282,24 @@ public function isEmpty() {
 		return empty( $this->statements );
 	}
 
+	/**
+	 * @since 3.0
+	 *
+	 * @param Statement $statement
+	 *
+	 * @return int|bool
+	 */
+	public function indexOf( Statement $statement ) {
+		$index = 0;
+
+		foreach ( $this->statements as $s ) {
+			if ( $s->equals( $statement ) ) {
```

Comment:
This new `indexOf` will fail if there is a statement with the same GUID but, for example, a reordered list of references.